app/liquid_tags/unified_embed.rb

```diff
@@ -0,0 +1,46 @@
+# The purpose of this singleton class is to provide a registry for the
+# numerous "link/embedded" type liquid tags.
+#
+# Each of those liquid tags must self-register their lookup regular
+# expression.
+class UnifiedEmbed
+  include Singleton
+
+  def self.current
+    instance
+  end
+
+  # @api public
+  #
+  # @param klass [Class] the LiquidTag class that we use when we match
+  #        the given :regexp
+  # @param regexp [Regexp] the regular expression that when matched
+  #        means we use the associated :klass
+  def self.register(klass, regexp:)
+    instance.register(klass, regexp: regexp)
+  end
+
+  # @api public
+  #
+  # @param link [String] the string that includes the URI for the
+  #        embed and possibly additional attributes, depending on how
+  #        the registered liquid tag parses this string.
+  # @return [Class] a descendant class of LiquidTagBase
+  def self.find_liquid_tag_for(link:)
+    instance.find_liquid_tag_for(link: link)
+  end
+
+  def initialize
+    @registry = []
+  end
+
+  def register(klass, regexp:)
+    @registry << [regexp, klass]
+  end
+
+  def find_liquid_tag_for(link:)
+    link = ActionController::Base.helpers.strip_tags(link).strip if link.include?("href=")
+    _regexp, klass = @registry.detect { |regexp, _tag_class| regexp.match?(link) }
```

Comment:
Rubocop complained about the use of `find` over `detect`; I investigated their differences and found no compelling explanation. 🤷🏾‍♀️  So I went with `detect`.

Reply:
@msarit `find` and `detect` have identical functionality. They're quite literally the same method. 😄

```ruby
[1] pry(main)> a = []; a.method(:find) == a.method(:detect)
=> true
```

  The `Enumerable` module has a few of those:

| Method | Alias |
|-|-|
| `find` | `detect` |
| `find_all` | `select` |
| `map` | `collect` |
| `reduce` | `inject` |

The names on the right come from Ruby's roots in [Smalltalk](https://en.wikipedia.org/wiki/Smalltalk) and the names on the left come from Lisp. Rubocop is trying to standardize on one of each which, honestly, I haven't found necessary. I used to have opinions on it but honestly I was just being pedantic and I think Rubocop is, too.

IIRC Rubocop also tries to standardize on other aliased methods like `count` vs `size` vs `length`, which are all equivalent in Ruby, as well, with the only difference being that `count` can take a block — it's basically `select { ... }.count` but without the intermediate array allocation. They have different behavior in ActiveRecord but tbh I think that does more harm than good because like 6 people in the world remember which is which.

Reply:
very helpful - thank you!!